### DISCUSSION TITLE
entgql: enable automatic field collection in generated resolvers

### DISCUSSION THREAD

**Discussion by shadowspore** [ent/ent#1451] [Q&A]
Is there any reason why generated resolvers does not call query.CollectFields(ctx)?

https://github.com/ent/contrib/blob/master/entgql/internal/todo/ent/edge.go#L21-L34

**Comment by a8m** [comment]
Tagging @alexsn, since he's the one that added this functionality.

**Comment by alexsn** [comment]
Field collection should occur on the query root field and be recursive from there. Field are not collected in the linked location as those are subfield resolvers.
